core/dbt/context/providers.py

```diff
@@ -1636,7 +1636,7 @@ def this(self) -> Optional[str]:
         if self.model.this_input_node_unique_id:
             this_node = self.manifest.expect(self.model.this_input_node_unique_id)
             self.model.set_cte(this_node.unique_id, None)  # type: ignore
-            return self.adapter.Relation.add_ephemeral_prefix(this_node.name)
+            return self.adapter.Relation.add_ephemeral_prefix(this_node.identifier)  # type: ignore
```

Comment:
The root causes of the regressions reported in https://github.com/dbt-labs/dbt-core/issues/10754 and https://github.com/dbt-labs/dbt-core/issues/10763 was that:

1. the update from `this_node.name` to `this_node.identifier` was accidentally overlooked during the implementation and code review of https://github.com/dbt-labs/dbt-core/pull/10290/files, and
2. there were no functional tests to catch those particular cases

This PR addresses _both_ of those root causes ✅ ✅